### PR TITLE
issue #312 append headers on emit instead of replacing them

### PR DESCRIPTION
**Pull request by frairon**

### CODE REVIEW

File: examples/4-tests/example_test.go [hunk 1]
```diff
@@ -353,10 +353,16 @@ func Test_7InputOutputWithHeaders(t *testing.T) {
 	proc, _ := goka.NewProcessor([]string{}, goka.DefineGroup("group",
 		goka.Input("input", new(codec.String), func(ctx goka.Context, msg interface{}) {
 			processorHeaders = ctx.Headers()
-			ctx.Emit("output", ctx.Key(), fmt.Sprintf("forwarded: %v", msg), goka.WithCtxEmitHeaders(
-				map[string][]byte{
-					"Header1": []byte("to output"),
-				}))
+			ctx.Emit("output", ctx.Key(), fmt.Sprintf("forwarded: %v", msg),
+				goka.WithCtxEmitHeaders(
+					map[string][]byte{
```
Comment (mrahs) [inline examples/4-tests/example_test.go:358]: Would be great to add colliding headers as well to assert overwrite behaviour, but not critical.

Comment (frairon) [reply examples/4-tests/example_test.go:358]: No problem, just added it.

File: examples/4-tests/example_test.go [hunk 2]
```diff
@@ -353,10 +353,18 @@ func Test_7InputOutputWithHeaders(t *testing.T) {
...
+						"Header1": []byte("to output"),
+						"Header2": []byte("to output2"),
+					}),
+				goka.WithCtxEmitHeaders(
+					map[string][]byte{
+						"Header2": []byte("to output3"),
+						"Header3": []byte("to output4"),
```
Comment (mmreza79) [inline examples/4-tests/example_test.go:365]: Is it header `Header2, Header3`, or `Header3 Header4`?

File: examples/4-tests/example_test.go [hunk 3]
```diff
@@ -376,6 +384,11 @@ func Test_7InputOutputWithHeaders(t *testing.T) {
 			"Header1": []byte("value 1"),
 			"Header2": []byte("value 2"),
 		}),
+		tester.WithHeaders(
+			map[string][]byte{
+				"Header2": []byte("value 3"),
+				"Header3": []byte("value 4"),
```
Comment (mmreza79) [inline examples/4-tests/example_test.go:390]: same question

File: examples/4-tests/example_test.go (hunk 2 above)
Comment (frairon) [reply examples/4-tests/example_test.go:365]: ok I admit the values are a bit misleading, but the idea here was to test the overwriting of header values. `Header2` is in both header-options and gets overwritten by the "last version".

File: examples/4-tests/example_test.go (hunk 3 above)
Comment (frairon) [reply examples/4-tests/example_test.go:390]: same answer :)

**Review by mmreza79 — APPROVED** [review]

**Pull request merged**